### PR TITLE
2920 Enhance padding mode for Tensor data

### LINKED ISSUE #2920: RandZoom change the input type
**Describe the bug**
When using `RandZoom` with the input type of `torch.Tensor` it will output `np.array`!

**To Reproduce**

```python
import torch
from monai.transforms import RandZoom
a = torch.randn(10, 3, 20, 20)
type(RandZoom(prob=1.0, min_zoom=0.9, max_zoom=1.1)(a))
```

**Expected behavior**
expected correct output:
```
<class 'torch.Tensor'>
```
but this will output:
```
<class 'numpy.ndarray'>
```

**Environment**
```
================================
Printing MONAI config...
================================
MONAI version: 0.6.0+117.g97fd2b92
Numpy version: 1.21.1
Pytorch version: 1.10.0a0+ecc3718
MONAI flags: HAS_EXT = False, USE_COMPILED = False
MONAI rev id: 97fd2b9201ea5e21692126539b0bddb99f1f7b23

Optional dependencies:
Pytorch Ignite version: 0.4.5
Nibabel version: 3.2.1
scikit-image version: 0.15.0
Pillow version: 8.2.0
Tensorboard version: 2.5.0
gdown version: 3.13.0
TorchVision version: 0.11.0a0
tqdm version: 4.53.0
lmdb version: 1.2.1
psutil version: 5.8.0
pandas version: 1.1.4
einops version: 0.3.0

For details about installing the optional dependencies, please visit:
    https://docs.monai.io/en/latest/installation.html#installing-the-recommended-dependencies


================================
Printing system config...
================================
System: Linux
Linux version: Ubuntu 20.04.2 LTS
Platform: Linux-5.8.0-59-generic-x86_64-with-glibc2.10
Processor: x86_64
Machine: x86_64
Python version: 3.8.10
Process name: python
Command: ['python', '-c', 'import monai; monai.config.print_debug_info()']
Open files: []
Num physical CPUs: 8
Num logical CPUs: 16
Num usable CPUs: 16
CPU usage (%): [1.4, 0.7, 96.5, 1.4, 3.5, 0.7, 2.1, 3.4, 0.7, 2.0, 0.7, 0.7, 0.0, 0.0, 0.0, 0.0]
CPU freq. (MHz): 1206
Load avg. in last 1, 5, 15 mins (%): [2.8, 3.6, 11.3]
Disk usage (%): 93.5
Avg. sensor temp. (Celsius): UNKNOWN for given OS
Total physical memory (GB): 31.1
Available memory (GB): 25.3
Used memory (GB): 5.3

================================
Printing GPU config...
================================
Num GPUs: 2
Has CUDA: True
CUDA version: 11.4
cuDNN enabled: True
cuDNN version: 8202
Current device: 0
Library compiled for CUDA architectures: ['sm_52', 'sm_60', 'sm_61', 'sm_70', 'sm_75', 'sm_80', 'sm_86', 'compute_86']
GPU 0 Name: NVIDIA TITAN RTX
GPU 0 Is integrated: False
GPU 0 Is multi GPU board: False
GPU 0 Multi processor count: 72
GPU 0 Total memory (GB): 23.7
GPU 0 CUDA capability (maj.min): 7.5
GPU 1 Name: NVIDIA TITAN RTX
GPU 1 Is integrated: False
GPU 1 Is multi GPU board: False
GPU 1 Multi processor count: 72
GPU 1 Total memory (GB): 23.7
GPU 1 CUDA capability (maj.min): 7.5
```

### PR DESCRIPTION
**Pull request by Nic-Ma**
Fixes #2920 .

### Description
This PR enhanced the padding modes and additional args for Tensor data.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Comment by Nic-Ma** [comment]
Hi @wyli and @rijobro ,

I enhanced the `padding mode` and `addtional args` in the `Pad` transform in this draft PR, could you please help review the proposal first?
If you agree with the proposal, I will try to update all the related places in the codebase in this PR.

Thanks in advance.

**Comment by rijobro** [comment]
It would certainly be great if we could use torch.pad for torch input and numpy.pad for numpy input. I think the reason I made these caveats was because the torch.pad is more limited compared to the numpy version. I suspect this will be reflected by the unit tests, but would be great if you could get it working!

**Comment by Nic-Ma** [comment]
Hi @rijobro ,

Thanks for your support, I think it's user's responsibility to choose the correct mode for the input data type(numpy or Tensor), we can explicitly show the supported modes of `numpy` and `Tensor` in the doc-string. If the passing mode is not compatible, it will raise an error naturally.

Thanks.

**Comment by Nic-Ma** [comment]
Hi @wyli ,

Do you have any other concerns?
If no, I will try to complete this PR for all the related components following this direction: numpy input -> np.pad, tensor input -> torch.pad, rename `np_kwargs` -> `kwargs`.

Thanks.


Thanks.

**Comment by wyli** [comment]
do we need a suitable default for both numpy and torch inputs?

**Comment by Nic-Ma** [comment]
> do we need a suitable default for both numpy and torch inputs?

Maybe let's change the default mode to `constant`? @wyli @rijobro What do you guys think?

Thanks.

**Comment by wyli** [comment]
constant 0 may 'break' the original intensity distribution and the results of mean/max/min would be very different from the original ones...I think 'edge'/'replicate' is a safer choice

**Comment by Nic-Ma** [comment]
Hi @wyli , @rijobro ,

I have updated the PR based on the previous discussion.
Could you please help review it?
I didn't update the transforms that don't support Tensor data yet.

Thanks in advance.

File: monai/utils/enums.py [hunk 1]
```diff
@@ -101,17 +112,6 @@ class BlendMode(Enum):
     GAUSSIAN = "gaussian"
 
 
-class PytorchPadMode(Enum):
-    """
```
Comment (wyli) [inline monai/utils/enums.py:105]: Looks like the changes in this file is not needed

File: monai/transforms/croppad/array.py [hunk 2]
```diff
@@ -61,60 +70,69 @@
 class Pad(Transform):
```
Comment (wyli) [inline monai/transforms/croppad/array.py:70]: this class is not in the docs/; could you please help add?

File: monai/utils/type_conversion.py [hunk 3]
```diff
@@ -214,4 +214,12 @@ def convert_to_dst_type(src: Any, dst: NdarrayOrTensor) -> Tuple[NdarrayOrTensor
     device = None
     if isinstance(dst, torch.Tensor):
         device = dst.device
-    return convert_data_type(data=src, output_type=type(dst), device=device, dtype=dst.dtype)
+
+    output_type: Any
```
Comment (wyli) [inline monai/utils/type_conversion.py:218]: could you please help update the docstring and the test `tests/test_convert_data_type.py`

File: monai/transforms/utils.py [hunk 4]
```diff
@@ -1259,5 +1263,31 @@ def print_table_column(name, torch, numpy, color=Colors.none):
...
+    """
+    Utility to convert padding mode between numpy array and PyTorch Tensor.
+
+    Args:
+        dst: target data to convert padding mode for, should be numpy array or PyTorch Tensor.
+        mode: current padding mode.
+
+    """
+    mode = mode.value if isinstance(mode, (NumpyPadMode, PytorchPadMode)) else mode
+    if isinstance(dst, torch.Tensor):
```
Comment (wyli) [inline monai/transforms/utils.py:1276]: this style could be improved according to https://deepsource.io/gh/Project-MONAI/MONAI/run/d1d80c47-af73-4ac8-86c2-d5e200f7a971/python/PYL-R1705

**Review by wyli — APPROVED** [review]
thanks, please see some minor comments inline

File: monai/utils/enums.py (hunk 1 above)
Comment (Nic-Ma) [reply monai/utils/enums.py:105]: Sure, restored in the latest commit.
Thanks.

File: monai/transforms/croppad/array.py (hunk 2 above)
Comment (Nic-Ma) [reply monai/transforms/croppad/array.py:70]: Added in the latest commit.
Thanks.

File: monai/utils/type_conversion.py (hunk 3 above)
Comment (Nic-Ma) [reply monai/utils/type_conversion.py:218]: updated in the latest commit.
Thanks.

File: monai/transforms/utils.py (hunk 4 above)
Comment (Nic-Ma) [reply monai/transforms/utils.py:1276]: Sure, improved in the latest commit.
Thanks.
